packages/concerto-core/lib/introspect/property.js

```diff
@@ -233,6 +233,14 @@ class Property extends Decorated {
     isPrimitive() {
         return ModelUtil.isPrimitiveType(this.getType());
     }
+
+    /**
+     * Returns true if this property is an aggregate type.
+     * @return {boolean} true if the property is an aggregate type.
+     */
+    isAggregate() {
```

Comment:
isRecord?

Reply:
I intended this to cover the case for `Set`, `Union`, `Array` in future too